flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/service/StandaloneFlinkServiceTest.java

```diff
@@ -99,9 +99,12 @@ public void testDeleteClusterDeploymentWithHADelete() throws Exception {
         List<Deployment> deployments = kubernetesClient.apps().deployments().list().getItems();
         assertEquals(2, deployments.size());
 
+        var requestsBeforeDelete = mockServer.getRequestCount();
         flinkStandaloneService.deleteClusterDeployment(
                 flinkDeployment.getMetadata(), flinkDeployment.getStatus(), configuration, true);
 
+        assertEquals(6, mockServer.getRequestCount() - requestsBeforeDelete);
```

Comment:
I don't really feel that this adequately tests the functionality. We should make sure that the deletion is blocked until the service returns empty list right?